filebeat/harvester/log.go

```diff
@@ -148,11 +160,12 @@ func (h *Harvester) shouldExportLine(line string) bool {
 // the file system is scanned
 func (h *Harvester) openFile() error {
 
-	f, err := file.ReadOpen(h.path)
+	f, err := file.ReadOpen(h.state.Source)
```

Comment:
You are right :-( This could be a reason we leak some file descriptors. We must backport this fix to 1.3 @tsg FYI